src/middlewares/setInstance.js

```diff
@@ -0,0 +1,18 @@
+const { logger } = require("../utils/logger");
+
+module.exports = function (instanceValue) {
+    return setInstance=(req, res, next) => {
+        if (req.headers['x-openhim-clientid']) {
+          const clientId = req.headers['x-openhim-clientid'];
+          if (clientId === 'echis') {
+            res.locals.instanceValue = "STAGING";
```

Comment:
Let's not define defaults here, the middleware should work with only what is passed in the request headers and fail if the header does not exist